Loading of data by using pandas library.
--

In [43]:
import pandas as pd
df=pd.read_csv("student-mat.csv",sep=";")

Analyzing of data by using pandas.This is very important step.To analyze the data we use function like
1).head():-starting rows
2).tail():-ending rows
3).info():-It gives all info of dataset
4).shape:-it give number of rows ad columns.
--

In [44]:
print(df.head(3))

  school sex  age address famsize Pstatus  Medu  Fedu     Mjob     Fjob  ...  \
0     GP   F   18       U     GT3       A     4     4  at_home  teacher  ...   
1     GP   F   17       U     GT3       T     1     1  at_home    other  ...   
2     GP   F   15       U     LE3       T     1     1  at_home    other  ...   

  famrel freetime  goout  Dalc  Walc health absences G1 G2  G3  
0      4        3      4     1     1      3        6  5  6   6  
1      5        3      3     1     1      3        4  5  5   6  
2      4        3      2     2     3      3       10  7  8  10  

[3 rows x 33 columns]


In [45]:
print(df.tail(3))

    school sex  age address famsize Pstatus  Medu  Fedu      Mjob     Fjob  \
392     MS   M   21       R     GT3       T     1     1     other    other   
393     MS   M   18       R     LE3       T     3     2  services    other   
394     MS   M   19       U     LE3       T     1     1     other  at_home   

     ... famrel freetime  goout  Dalc  Walc health absences  G1  G2  G3  
392  ...      5        5      3     3     3      3        3  10   8   7  
393  ...      4        4      1     3     4      5        0  11  12  10  
394  ...      3        2      3     3     3      5        5   8   9   9  

[3 rows x 33 columns]


In [46]:
print(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 395 entries, 0 to 394
Data columns (total 33 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   school      395 non-null    object
 1   sex         395 non-null    object
 2   age         395 non-null    int64 
 3   address     395 non-null    object
 4   famsize     395 non-null    object
 5   Pstatus     395 non-null    object
 6   Medu        395 non-null    int64 
 7   Fedu        395 non-null    int64 
 8   Mjob        395 non-null    object
 9   Fjob        395 non-null    object
 10  reason      395 non-null    object
 11  guardian    395 non-null    object
 12  traveltime  395 non-null    int64 
 13  studytime   395 non-null    int64 
 14  failures    395 non-null    int64 
 15  schoolsup   395 non-null    object
 16  famsup      395 non-null    object
 17  paid        395 non-null    object
 18  activities  395 non-null    object
 19  nursery     395 non-null    object
 20  higher    

In [47]:
print(df.shape)

(395, 33)


In [48]:
print(df.isnull())

     school    sex    age  address  famsize  Pstatus   Medu   Fedu   Mjob  \
0     False  False  False    False    False    False  False  False  False   
1     False  False  False    False    False    False  False  False  False   
2     False  False  False    False    False    False  False  False  False   
3     False  False  False    False    False    False  False  False  False   
4     False  False  False    False    False    False  False  False  False   
..      ...    ...    ...      ...      ...      ...    ...    ...    ...   
390   False  False  False    False    False    False  False  False  False   
391   False  False  False    False    False    False  False  False  False   
392   False  False  False    False    False    False  False  False  False   
393   False  False  False    False    False    False  False  False  False   
394   False  False  False    False    False    False  False  False  False   

      Fjob  ...  famrel  freetime  goout   Dalc   Walc  health  absences  \

In [49]:
print(df.isnull().sum())

school        0
sex           0
age           0
address       0
famsize       0
Pstatus       0
Medu          0
Fedu          0
Mjob          0
Fjob          0
reason        0
guardian      0
traveltime    0
studytime     0
failures      0
schoolsup     0
famsup        0
paid          0
activities    0
nursery       0
higher        0
internet      0
romantic      0
famrel        0
freetime      0
goout         0
Dalc          0
Walc          0
health        0
absences      0
G1            0
G2            0
G3            0
dtype: int64


In [50]:
print(df.duplicated())

0      False
1      False
2      False
3      False
4      False
       ...  
390    False
391    False
392    False
393    False
394    False
Length: 395, dtype: bool


In [51]:
print(df.duplicated().sum())

0


By analyzing and cleaning data we understand that.
1)Data is alredy cleaned.
2)No duplicated
3)their are total 33 columns and 395 rows.
--

Anylizing Questions.
--

Average final grade (G3).
-->10.415189873417722
--

In [52]:
print(df["G3"].mean())

10.415189873417722


How many students scored above 15?
-->40
--

In [53]:
(df["G3"] > 15).sum()

np.int64(40)

Is study time correlated with performance?
-->0.09781968965319636
--

In [54]:
df["studytime"].corr(df["G3"])

np.float64(0.09781968965319636)

Which gender performs better on average?
-->Male gender performance is better on average.
   Female:-9.966346
   Male:-10.914439
   --

In [ ]:
df.groupby("sex")["G3"].mean()

Visualizations..
Visualization is used for good understanding of data by graphs.
--

Here we use subplots to plot histograms on same figure for grade G1,G2,G3.
--

In [ ]:
import matplotlib.pyplot as plt
plt.figure(figsize=(12, 4))
G1=df["G1"]
plt.subplot(1,3,1)
plt.hist(G1,bins=5,edgecolor="black")
plt.title("G1")
G2=df["G2"]
plt.subplot(1,3,2)
plt.hist(G2,bins=5,edgecolor="black")
plt.title("G2")
G3=df["G3"]
plt.subplot(1,3,3)
plt.hist(G3,bins=5,edgecolor="black")
plt.title("G3")
plt.tight_layout()

Scatter plot for grade G1,G2,G3.
--

In [ ]:
import matplotlib.pyplot as plt
plt.figure(figsize=(12, 4))
s_time = df["studytime"]
G1 = df["G1"]
G2 = df["G2"]
G3 = df["G3"]
plt.scatter(s_time, G1, label="G1", alpha=0.7)
plt.scatter(s_time, G2, label="G2", alpha=0.7)
plt.scatter(s_time, G3, label="G3", alpha=0.7)
plt.xlabel("Study Time")
plt.ylabel("Grades")
plt.title("Study Time vs Student Performance")
plt.legend()
plt.tight_layout()
plt.show()

Average of G1,G2,G3 by comparing Male and Female..
--

In [ ]:
import matplotlib.pyplot as plt
plt.figure(figsize=(12, 4))

plt.subplot(1, 3, 1)
g1 = df.groupby("sex")["G1"].mean()
plt.bar(g1.index, g1.values, edgecolor="black")
plt.title("Average G1")
plt.xlabel("Sex")
plt.ylabel("Average Score")

plt.subplot(1, 3, 2)
g2 = df.groupby("sex")["G2"].mean()
plt.bar(g2.index, g2.values, edgecolor="black")
plt.title("Average G2")
plt.xlabel("Sex")

plt.subplot(1, 3, 3)
g3 = df.groupby("sex")["G3"].mean()
plt.bar(g3.index, g3.values, edgecolor="black")
plt.title("Average G3")
plt.xlabel("Sex")
plt.tight_layout()
plt.show()
